vpr/src/route/route_common.cpp

```diff
@@ -1114,15 +1117,46 @@ static vtr::vector<ClusterBlockId, std::vector<int>> load_rr_clb_sources(const t
     return rr_blk_source;
 }
 
+static vtr::vector<ClusterNetId, uint8_t> load_is_clock_net() {
+    vtr::vector<ClusterNetId, uint8_t> is_clock_net;
+
+    auto& atom_ctx = g_vpr_ctx.atom();
+    std::set<AtomNetId> clock_nets = find_netlist_physical_clock_nets(atom_ctx.nlist);
+
+    auto& cluster_ctx = g_vpr_ctx.clustering();
+    auto nets = cluster_ctx.clb_nlist.nets();
+
+    is_clock_net.resize(nets.size());
+    for (auto net_id : nets) {
+        is_clock_net[net_id] = clock_nets.find(atom_ctx.lookup.atom_net(net_id)) != clock_nets.end();
+    }
+
+    return is_clock_net;
+}
+
 vtr::vector<ClusterNetId, t_bb> load_route_bb(int bb_factor) {
     vtr::vector<ClusterNetId, t_bb> route_bb;
 
     auto& cluster_ctx = g_vpr_ctx.clustering();
+    auto& route_ctx = g_vpr_ctx.routing();
+
+    t_bb full_device_bounding_box;
+    {
+        auto& device_ctx = g_vpr_ctx.device();
+        full_device_bounding_box.xmin = 0;
+        full_device_bounding_box.ymin = 0;
+        full_device_bounding_box.xmax = device_ctx.grid.width() - 1;
+        full_device_bounding_box.ymax = device_ctx.grid.height() - 1;
+    }
 
     auto nets = cluster_ctx.clb_nlist.nets();
     route_bb.resize(nets.size());
     for (auto net_id : nets) {
-        route_bb[net_id] = load_net_route_bb(net_id, bb_factor);
+        if (!route_ctx.is_clock_net[net_id]) {
```

Comment:
Please add a comment on why you're doing this (well explained in the PR, but put an overview here too).
Is is_clock_net true for only globally routed clocks (special network) or all clocks including locally routed clocks?  Ideally we'd turn off the bounding box only for clocks that are being routed on special global networks.

Reply:
> Is is_clock_net true for only globally routed clocks (special network) or all clocks including locally routed clocks? Ideally we'd turn off the bounding box only for clocks that are being routed on special global networks.

Any net that sinks at a `<clock>` is consider a clock net.